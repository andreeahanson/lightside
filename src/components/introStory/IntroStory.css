aside {
  width: 20%;
  overflow: hidden;
}

.slider {
  animation: slide-up 15s linear infinite;
}

@keyframes slide-up {
  from {
      transform: translateY(600px);
  }
  to {
      transform: translateY(-100px);
  }
}