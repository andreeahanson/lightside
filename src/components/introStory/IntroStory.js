import React from 'react';
import './IntroStory.css';

const IntroStory = ({ films }) => {
  const movie = films[Math.floor(Math.random() * films.length)] || {};
  return (
    <aside>
      <p>{movie.opening_crawl}</p>
      <h3>{movie.title}</h3>
      <h4>{movie.release_date}</h4>
    </aside>
  );
};

export default IntroStory;
