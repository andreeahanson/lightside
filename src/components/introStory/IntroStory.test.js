import React from 'react';
import { shallow } from 'enzyme';
import IntroStory from './IntroStory';
import sampleFilms from '../../sampleDataDirectory/sampleFilms'

describe('IntroStory', () => {
  let wrapper;
  
    beforeEach(() => {
      wrapper = shallow(<IntroStory 
      films ={sampleFilms.results}
      />)
    })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })


})