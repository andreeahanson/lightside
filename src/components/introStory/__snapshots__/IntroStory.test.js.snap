// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroStory should match the snapshot 1`] = `ShallowWrapper {}`;
